pandas/core/frame.py

```diff
@@ -7898,6 +7915,13 @@ def _get_data(axis_matters):
                 raise NotImplementedError(msg)
             return data
 
+        if axis == 0 and column_wise:
+            # column-wise reduction
+            df = self
+            if numeric_only is True:
+                df = _get_data(axis_matters=True)
+            return DataFrame._reduce_columns(df, op, name, skipna=skipna, **kwds)
```

Comment:
is this different from `df._reduce_columns(op, name, skipna=skipna, **kwds)`?

Reply:
Ah, no, of course not, that's an oversight